sbws/core/scanner.py

```diff
@@ -131,7 +132,25 @@ def measure_bandwidth_to_server(session, conf, dest, content_length):
     return results
 
 
+def obtain_ed25519(controller, relay):
+    """Obtain ed25519 identity of the relay. Modifies relay object.
+
+    :param stem.control.Controller controller: stem Tor _init_controller
+    :param Relay: the relay
+    """
+    if relay.identifier is None:
+        try:
+            log.debug('Getting microdescriptor to obtain ed25519 identity.')
+            mdesc = controller.get_microdescriptor(relay.fingerprint)
+            if 'ed25519' in mdesc.identifiers.keys():
+                relay.identifier_type = 'ed25519'
+                relay.identifier = mdesc.identifiers['ed25519']
+        except DescriptorUnavailable as e:
+            log.debug('e: %s')
+
+
 def measure_relay(args, conf, destinations, cb, rl, relay):
+    obtain_ed25519(cb.controller, relay)
```

Comment:
Please also try to obtain the ed25519 id when the relay object is created.

Reply:
I wanted to do like that, but Relay object is only created at the end, when dumping results, as part of Result object: https://github.com/pastly/simple-bw-scanner/blob/master/sbws/lib/resultdump.py#L150
The first place where 1st is obtained descriptors (not relays) is in https://github.com/pastly/simple-bw-scanner/blob/master/sbws/lib/relaylist.py#L134
Without refactoring code, 2 possible solutions:
- obtain ed25519 when descriptors are obtained, and modify descriptor object
- obtain ed25519 in Result and/or Relay object
Which one seems better?

Reply:
or other ideas where to do it? (@pastly)